### ISSUE TITLE
Add Support for Pro Level J-K-L Key Bindings

### ISSUE THREAD

**Issue by jckstudios** [iina/iina#3224]
**What you want IINA to do:**
As a professional video editor, I'm very used to using the JKL keys as shuttle control for playback. 

One press of L plays back 1x, a second press goes 2x, and a third 4x etc. The J key steps those speed increases down, or if starting from pause it does the inverse i.e., first press -1x, second -2x, third -4x etc. The K key always stops playback.

**What IINA does currently:**
With the current key bindings options there is no way to replicate this functionality. You have to set it to seek ahead x-number of frames, missing the in-between frames. 

**Why you think this should be added:**
For people who live and breathe minute control of video playback, the JKL shuttle is a standard across virtually all workflows and is a second nature for many. As such it is hard to use IINA at present on day to day projects, as such things as opening and seeking through a file can be frustrating. I think it would be a great addition to IINA and make it even more attractive to those who work in post-production.

**Examples of other projects that have something similar:**
[Premiere Pro](https://nofilmschool.com/2018/10/how-use-j-k-and-l-keys-premiere-pro-speed-your-workflow), [Final Cut Pro X](https://www.premiumbeat.com/blog/fcp-super-shortcuts-turn-your-keyboard-into-a-jog-shuttle/), [AVID](http://resources.avid.com/SupportFiles/attach/Interplay_Central/IPC_Help/toc.html#page/InterplayCentral_Help/UG_Media.16.09.html), [DaVinci Resolve](https://forum.blackmagicdesign.com/viewtopic.php?f=21&t=104959), [Quicktime](https://sites.duke.edu/ddmc/2013/10/31/quicktime-j-k-l-keyboard-shortcuts/) [etc](https://www.premiumbeat.com/blog/video-editing-j-k-l-shortcuts/).

**Comment by daniel-toman** [comment]
Yes please! IINA is a better Mac video player for video editors than Quicktime Player or VLC. Quicktime Player interprets color metadata differently than web video hosting services, so color in Quicktime Player will not necessarily look like color on Vimeo / YouTube etc. IINA color rendering matches web video hosting color rendering. And IINA is capable of playing back more video formats and damaged video files better than Quicktime Player. IINA's UI and ability to fast forward / rewind is better than VLC's. 

If IINA would replicate the J-K-L key functionality of Quicktime Player and all major video editing software, it would be ideal for video editor use! Thanks for this great tool.

**Comment by svobs** [comment]
It looks like #3677 is a duplicate of this.

It looks like "JKL" functionality is more of an informal convention than a real standard. Some quick Internet searches dug up [this](https://resources.avid.com/SupportFiles/attach/Interplay_Central/IPC_Help/toc.html#page/InterplayCentral_Help/UG_Media.16.09.html) and [this](https://sites.duke.edu/ddmc/2013/10/31/quicktime-j-k-l-keyboard-shortcuts/)

### Common principles:
- K key stops playback
- J goes backward at increasing intervals. L goes forward at increasing speeds. The speeds aren't standardized

### QuickTime's behavior:
- 'L' key repeated presses = 1x -> 2x -> 5x -> 10x -> 30x -> 60x -> (loop back to) 2x -> 5x...
- 'J' key repeated presses = -1x -> -2x -> -5x -> -10x -> -30x -> -60x -> (loop back to) -2x > -5x...

### Various players (not QuickTime): alternative or other functions:
1. Do not accelerate; e.g., 'L' key repeated presses = 1x -> 2x -> 3x -> 4x
2. Accelerate but stop at a maximum rather than looping ; e.g., 'L' key repeated presses = 1x -> 2x -> 5x -> 10x -> 30x -> 30x...
3. Slow-motion: Press the K and L keys together for slow forward at ¼ times normal speed. Press the K and J keys together for slow backward at ¼ times normal speed.
4. Frame step: Press and hold the K key and tap the L key or the J key to step through footage one frame at a time.

You can approximate some of this functionality by mapping MPV's keys like this:
```
k set speed 1.0; set pause yes
l multiply speed 2.0; set pause no
```

MPV only started supporting backwards playback in the 0.30 release, and it seems to be pretty half-spirited. It requires `-vf “reverse”` to be specified when the file is opened; toggling it during playback is not supported. So unfortunately, doing a "rewind" feature seems to be out of the question.

I was prepared to recommend adding special UI options for this functionality, but given MPV's lack of support for rewind, it's not clear the user experience would be any good.

**Comment by daniel-toman** [comment]
Thanks for the reply, the limited backwards playback functionality is understandable, VLC also can't play in reverse I think. I think implementing K and L would be great as you describe, with J maybe skipping back 1 second or whatever the closest analogy to rewinding could be. In the future if J could do backwards playback as that functionality matures, that would be ideal.

JKL keys for video playback are like WASD keys for gaming, it's a way many people automatically expect to navigate even if other navigation methods are also available.

Thanks for describing the typical functionality in more detail, I'd mention that not only Avid Media Composer and Quicktime use the JKL keys for playback control, but also Adobe [Premiere Pro](https://helpx.adobe.com/premiere-pro/using/keyboard-shortcuts.html), [Davinci Resolve](https://documents.blackmagicdesign.com/SupportNotes/DaVinci_Resolve_11_Mac_Keyboard_Shortcuts.pdf), [Final Cut Pro X](https://support.apple.com/guide/final-cut-pro/keyboard-shortcuts-ver90ba5929/mac), basically all major Non Linear Editing software. Even editor-focused web apps like [Frame.io](https://support.frame.io/en/articles/7785-keyboard-shortcuts) also use JKL shortcut keys.

[People](https://twitter.com/avid/status/1329892786242547712) even have JKL [tattoos](https://www.instagram.com/p/CJYoc7VjU0U/) :)

**Comment by EugeneTarasenko** [comment]
I second this!
